platforms/software/ivy/src/main/java/org/gradle/api/publish/ivy/tasks/GenerateIvyDescriptor.java

```diff
@@ -41,77 +46,34 @@
 @UntrackedTask(because = "Gradle doesn't understand the data structures")
 public abstract class GenerateIvyDescriptor extends DefaultTask {
 
-    private Transient.Var<IvyModuleDescriptorSpec> descriptor = Transient.varOf();
-    private final Cached<IvyDescriptorFileGenerator.DescriptorFileSpec> ivyDescriptorSpec = Cached.of(this::computeIvyDescriptorFileSpec);
-
-    private Object destination;
-
-    @Inject
-    protected PathToFileResolver getFileResolver() {
-        throw new UnsupportedOperationException();
-    }
+    private final Transient<Property<IvyModuleDescriptorSpec>> descriptor = Transient.of(getObjectFactory().property(IvyModuleDescriptorSpec.class));
+    private final Provider<IvyDescriptorFileGenerator.DescriptorFileSpec> ivyDescriptorSpec = getDescriptor().map(descriptor ->
+        IvyDescriptorFileGenerator.generateSpec(toIvyModuleDescriptorInternal(descriptor))
+    );
 
     /**
      * The module descriptor metadata.
-     *
-     * @return The module descriptor.
      */
     @Internal
-    @ToBeReplacedByLazyProperty
-    public IvyModuleDescriptorSpec getDescriptor() {
+    @ReplacesEagerProperty
+    public Property<IvyModuleDescriptorSpec> getDescriptor() {
```

Comment:
🤔 Do we want to have this as a `Property`? It looks like a nested type 
/cc @jvandort